src/Integration.Vsix/Suppression/SuppressionManager.cs

```diff
@@ -0,0 +1,131 @@
+                SetupSuppressionHandling();
+            }
+            else
+            {
+                CleanupSuppressionHandling();
+            }
+        }
+
+        private void SetupSuppressionHandling()
+        {
+            liveIssueFactory = new LiveIssueFactory(serviceProvider);
+            delegateInjector = new DelegateInjector(ShouldIssueBeSuppressed);
+            serverIssueProvider = new ServerIssuesProvider(serviceProvider);
+        }
+
+        private void CleanupSuppressionHandling()
+        {
+            delegateInjector?.Dispose();
+            delegateInjector = null;
+            liveIssueFactory = null;
+            serverIssueProvider?.Dispose();
+            serverIssueProvider = null;
+        }
+
+        private void OnSolutionBindingChanged(object sender, bool e)
+        {
+            RefreshSuppresionHandling();
+        }
+
+        private bool ShouldIssueBeSuppressed(Diagnostic diagnostic)
+        {
+            // This method is called for every analyzer issue that is raised so it should be fast.
+            if (!diagnostic.Location.IsInSource) { return false; }
+            if (activeSolutionBoundTracker == null || !activeSolutionBoundTracker.IsActiveSolutionBound) { return false; }
+
+            LiveIssue liveIssue = liveIssueFactory.Create(diagnostic);
+            if (liveIssue == null)
+            {
+                return false; // Unable to get the data required to map a Roslyn issue to a SonarQube issue
+            }
+
+            // Issues match if:
+            // 1. Same component, same file, same error code, same line hash        // tolerant to line number changing
+            // 2. Same component, same file, same error code, same line             // tolarant to code on the line changing e.g. var rename
+
+            // TODO: ?need to make file path relative to the project file path
+            // As a minimum, the project, file and rule id must match
+            var issuesInFile = serverIssueProvider.GetServerIssues(liveIssue.ProjectId, liveIssue.IssueFilePath)
+                    .Where(i => StringComparer.OrdinalIgnoreCase.Equals(liveIssue.Diagnostic.Id, i.RuleKey)); // TODO: rule repository?
+
+            return issuesInFile.Any(i =>
+                    liveIssue.StartLine == i.Line ||
+                    StringComparer.Ordinal.Equals(liveIssue.LineChecksum, i.Checksum));
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        private void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    CleanupSuppressionHandling();
+                    activeSolutionBoundTracker.SolutionBindingChanged -= OnSolutionBindingChanged;
+                }
+
+                disposedValue = true;
+            }
+        }
+
+        // This code added to correctly implement the disposable pattern.
```

Comment:
Same comment as above - I'd remove comments from lines 123 and 126.

Reply:
Ok, but out of interest why does the presence of the comments make a difference to the analyzer?

Reply:
I meant if somebody changes the code (not the comments) not to follow the correct Dispose pattern, that would be picked up by the analyzer.